### PR TITLE
Added brave vpn feature flags to brave://flags and stub interface to BraveVpnService

### PR DESCRIPTION
**Pull request by simonhong**
* Added brave vpn feature flags to brave://flags
* Added stub interface for managing vpn connection status to `BraveVpnService`

Resolves https://github.com/brave/brave-browser/issues/17416
Resolves https://github.com/brave/brave-browser/issues/17417

Screen Shot 2021-08-10 at 10 57 01 AM

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Launch Nightly with clean profile
2. Enable vpn feature from brave://flags
3. Relaunch and check vpn button is visible on toolbar

### CODE REVIEW

File: components/brave_vpn/brave_vpn_service.h [hunk 1]
```diff
@@ -72,6 +88,8 @@ class BraveVpnService : public KeyedService {
       const std::string& body,
       const base::flat_map<std::string, std::string>& headers);
 
+  bool connected_ = false;
```
Comment (goodov) [inline components/brave_vpn/brave_vpn_service.h:91]: ```is_connected_```?

File: components/brave_vpn/brave_vpn_service.cc [hunk 2]
```diff
@@ -136,6 +137,37 @@ void BraveVpnService::VerifyPurchaseToken(ResponseCallback callback,
...
+
+bool BraveVpnService::IsConnected() const {
+  NOTIMPLEMENTED();
+
+  // Just return fake statue now.
+  return connected_;
+}
+
+void BraveVpnService::AddObserver(Observer* observer) {
+  observers_.AddObserver(observer);
```
[the remark this answers is not in the document]
Comment (simonhong) [reply components/brave_vpn/brave_vpn_service.cc:164]: Implemented `Shutdown()` in `BraveVpnService`. but nothing to cleanup now.

File: chromium_src/chrome/browser/about_flags.cc [hunk 3]
```diff
@@ -40,6 +41,19 @@ using ntp_background_images::features::kBraveNTPSuperReferralWallpaper;
...
 
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/components/brave_vpn/features.h"
+
+#define BRAVE_VPN_FEATURE_ENTRIES \
+    {"brave_vpn",                                                            \
+     flag_descriptions::kBraveVPNName,                                       \
+     flag_descriptions::kBraveVPNDescription,                                \
+     kOsMac | kOsWin | kOsAndroid,                                           \
+     FEATURE_VALUE_TYPE(brave_vpn::kBraveVPNFeature)},
```
[the remark this answers is not in the document]
Comment (simonhong) [reply chromium_src/chrome/browser/about_flags.cc]: Done. changed to `brave_vpn::features::kBraveVPN`.

File: components/brave_vpn/brave_vpn_service.h (hunk 1 above)
Comment (simonhong) [reply components/brave_vpn/brave_vpn_service.h]: fixed.

File: browser/ui/views/toolbar/brave_vpn_button.cc [hunk 4]
```diff
@@ -47,9 +48,13 @@ class BraveVPNButtonHighlightPathGenerator
...
 }  // namespace
 
-BraveVPNButton::BraveVPNButton()
+BraveVPNButton::BraveVPNButton(BraveVpnService* service)
     : ToolbarButton(base::BindRepeating(&BraveVPNButton::OnButtonPressed,
-                                        base::Unretained(this))) {
+                                        base::Unretained(this))),
+      service_(service) {
+  DCHECK(service_);
+  observation_.Observe(service);
```
[the remark this answers is not in the document]
Comment (simonhong) [reply browser/ui/views/toolbar/brave_vpn_button.cc]: Fixed.

File: components/brave_vpn/brave_vpn_service.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (goodov) [reply components/brave_vpn/brave_vpn_service.cc:166]: ```observers_.Clear();```?

[the remark this answers is not in the document]
Comment (simonhong) [reply components/brave_vpn/brave_vpn_service.cc:168]: Added.

**Review by goodov — APPROVED** [review]

**Pull request merged**